src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -0,0 +1,484 @@
+                return;
+            }
+
+            processJibriIqFromMeet(iq, chatMember);
+        }
+    }
+
+    private void processJibriIqFromMeet(JibriIq iq, XmppChatMember sender)
+    {
+        JibriIq.Action action = iq.getAction();
+
+        if (JibriIq.Action.UNDEFINED.equals(action))
+            return;
+
+        String senderMucJid = sender.getContactAddress();
+
+        logger.debug(
+                "Jibri request from " + senderMucJid + " iq: " + iq.toXML());
+
+        // start ?
+        if (JibriIq.Action.START.equals(action) &&
+            JibriIq.Status.OFF.equals(jibriStatus) &&
+            recorderComponentJid == null)
+        {
+            if (!verifyModeratorRole(iq))
+            {
+                logger.warn(
+                        "Ignored Jibri request from non-moderator: "
+                            + senderMucJid);
+                return;
+            }
+
+            // Check if we have Jibri available
+            String jibriJid = jibriDetector.selectJibri();
+            if (jibriJid == null)
+            {
+                sendErrorResponse(
+                    iq, XMPPError.Condition.service_unavailable, null);
+                return;
+            }
+
+            JibriIq startIq = new JibriIq();
+            startIq.setTo(jibriJid);
+            startIq.setType(IQ.Type.SET);
+            startIq.setAction(JibriIq.Action.START);
+
+            startIq.setStreamId(iq.getStreamId());
+            startIq.setUrl(iq.getUrl());
+
+            logger.debug("Starting Jibri recording: " + startIq.toXML());
+
+            IQ startReply
+                = (IQ) xmpp.getXmppConnection()
+                        .sendPacketAndGetReply(startIq);
+
+            logger.debug("Start response: " + startReply.toXML());
+
+            if (startReply == null)
+            {
+                sendErrorResponse(
+                        iq, XMPPError.Condition.request_timeout, null);
+                return;
+            }
+
+            if (IQ.Type.RESULT.equals(startReply.getType()))
+            {
+                recorderComponentJid = jibriJid;
+
+                setJibriStatus(JibriIq.Status.PENDING);
+
+                sendResultResponse(iq);
+                return;
+            }
+            else
+            {
+                XMPPError error = startReply.getError();
+                if (error == null)
+                {
+                    error = new XMPPError(
+                            XMPPError.Condition.interna_server_error);
+                }
+                sendPacket(IQ.createErrorResponse(iq, error));
+                return;
+            }
+        }
+        // stop ?
+        else if (JibriIq.Action.STOP.equals(action) &&
+            recorderComponentJid != null &&
+            (JibriIq.Status.ON.equals(jibriStatus) ||
+             JibriIq.Status.PENDING.equals(jibriStatus)))
+        {
+            if (!verifyModeratorRole(iq))
+                return;
```

Comment:
Maybe we should send a response IQ here? And we could move this to the beginning of the method and remove the duplicate check.